File: UmaMusumeTrpg/Models/System/List/Item.cs
using UmaMusumeTrpg.Entitys;
using UmaMusumeTrpg.Enum;
using UmaMusumeTrpg.Models.Base;

namespace UmaMusumeTrpg.Models.System.List
{

    public class Item : BaseItem
    {
        public Item(int id, string name, SysPermission sysPermission, UmaMusumeTrpgPermission umaMusumeTrpgPermission) : base(id)
        {
            Name = name;
            SysPermission = sysPermission;
            UmaMusumeTrpgPermission = umaMusumeTrpgPermission;
        }
        public Item(User user) : base(user.ID)
        {
            Name = user.Name;
            SysPermission = user.SysPermission;
            UmaMusumeTrpgPermission = user.UmaMusumeTrpgPermission;
        }

        public string Name { get; set; }

        public SysPermission SysPermission { get; set; }
        public UmaMusumeTrpgPermission UmaMusumeTrpgPermission { get; set; }
    }
}

File: Client/src/app/system/list/models/item.ts
import { SysPermission } from "../../../../lib/enum/sys-permission";
import { UmaMusumeTrpgPermission } from "../../../../lib/enum/uma-musume-trpg-permission";
import { BaseListItemModel } from "../../../../lib/models/list/base-list-item";

export interface ItemModel extends BaseListItemModel {
  name: string;
  sysPermission: SysPermission;
  umaMusumeTrpgPermission: UmaMusumeTrpgPermission;
}
